src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1600,6 +1609,12 @@ public bool IsDevelopmentMode()
         {
             return _runtimeConfigProvider.GetConfig().Runtime.Host.Mode is HostMode.Development;
         }
+
+        /// <inheritdoc />
+        public string GetDatabaseSourceName()
+        {
+            return _dataSourceName;
```

Comment:
If you have a getter for dataSourceName why not introduce a property with public `get` and `init` at constructor time?

Reply:
This function is not needed. I have removed it.